### PR TITLE
Fix decay rate in fine tuning of batch normalization

### LINKED ISSUE #1091: Bug in batch normalization when finetune = True?
Correct me if I'm wrong, but from my understanding, if the option `finetune` is `True`, we want to compute the population statistics of the training data so we can use this mean/var during testing.
In the current implementation, once the finetune mode is set, `decay` gets smaller the more training batches we see. However, the code weights the accumulated mean/var with `decay` and the current mean/var with `1-decay`.
Hence, if `N` gets very big, `self.avg_mean`/`self.avg_var` are almost equal to the mean/var of the current batch. It should be the other way around, e.g. the more data we saw, the less influence should the current mean/var have on the average statistics.

TD;LR `decay = 1. / self.N` should be replaced by `decay = 1. - 1. / self.N`

### PR DESCRIPTION
**Pull request by delta2323**
Fix #1091

### CODE REVIEW

**Comment by okuta** [comment]
Can you write a test?

**Comment by delta2323** [comment]
@okuta I added unit tests.

**Comment by okuta** [comment]
LGTM!
